server/controllers/profile.js

```diff
@@ -54,59 +68,58 @@ exports.updateProfile = asyncHandler(async (req, res, next) => {
       profilePic,
       description,
       availability,
-    } = req.body;
-
-    const profile = await Profile.findOneAndUpdate(
-      { user: id },
-      {
-        firstName,
-        lastName,
-        gender,
-        birthday,
-        email,
-        phoneNumber,
-        location,
-        profilePic,
-        description,
-        availability,
-      },
-      { new: true }
-    );
+    },
+    { new: true }
+  );
 
-    res.status(200).json({
-      profile,
-    });
+  res.status(200).json({
+    profile,
+  });
 });
 
 // @route GET /profile
 // @desc gets a profile with the given ID
 // @access Private
 exports.getProfile = asyncHandler(async (req, res, next) => {
-    const id = req.user._id;
-    const profile = await Profile.findOne({ user: id });
+  const id = req.user._id;
+  const profile = await Profile.findOne({ user: id });
 
-    if (!profile) {
-      res.status(404);
-      throw new Error("The profile does not exist");
-    }
+  if (!profile) {
+    res.status(404);
+    throw new Error("The profile does not exist");
+  }
 
-    res.status(200).json({
-      profile,
-    });
+  res.status(200).json({
+    profile,
+  });
 });
 
 // @route GET /profiles
 // @desc gets all profiles
 // @access Private
 exports.getAllProfiles = asyncHandler(async (req, res, next) => {
-    const profiles = await Profile.find();
+  const { search, start, end } = req.query;
 
-    if (!profiles) {
-      res.status(500);
-      throw new Error("0 results");
-    }
+  if (search != "" || start != "null" || end != "null") {
+    const profiles = await Profile.find({
+      location: { $regex: search, $options: "i" },
+      isSitter: true,
+    });
 
+    //TODO: Need to check dates to be valid
     res.status(200).json({
       profiles,
     });
+    return;
+  }
+
+  const profiles = await Profile.find();
+  if (!profiles) {
+    res.status(500);
+    throw new Error("0 results");
+  }
```

Comment:
So no need for this. Send the returned array to the client and let him figure out if it's empty.